### PR TITLE
[#3555] feat(spark-connector):  support multi scala version

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
support multi scala version 

### Why are the changes needed?

Fix: #3555 

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
1. existing IT and UT
2. in my local machine, tested spark sqls on spark3.4-2.13 and spark3.5-2.13

### CODE REVIEW

File: .github/workflows/spark-integration-test.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ jobs:
       matrix:
         architecture: [linux/amd64]
         java-version: [ 8, 11, 17 ]
+        scala-version: [ 2.12, 2.13 ]
```
Comment (FANNG1) [inline .github/workflows/spark-integration-test.yml:60]: will change to 2.12 when every thing is ready to merge

File: spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java [hunk 2]
```diff
@@ -239,6 +239,7 @@ public org.apache.spark.sql.connector.expressions.Transform[] toSparkTransform(
     return sparkTransforms.toArray(new org.apache.spark.sql.connector.expressions.Transform[0]);
   }
 
+  @SuppressWarnings("deprecation")
```
Comment (FANNG1) [inline spark-connector/spark-common/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java:242]: `JavaConverters.seqAsJavaList` is deprecated in scala 2.13

**Comment by FANNG1** [comment]
@jerryshao @qqqttt123 please help to review when you have time

File: .github/workflows/spark-integration-test.yml (hunk 1 above)
Comment (jerryshao) [reply .github/workflows/spark-integration-test.yml:60]: I think 2.12 only should be enough, otherwise, we will have too many CI pipeline.

**Comment by jerryshao** [comment]
Generally LGTM, @FANNG1 can you address one comment I mentioned above.

File: .github/workflows/spark-integration-test.yml (hunk 1 above)
Comment (FANNG1) [reply .github/workflows/spark-integration-test.yml]: done

**Review by jerryshao — DISMISSED** [review]

File: .github/workflows/spark-integration-test.yml [hunk 3]
```diff
@@ -91,9 +92,11 @@ jobs:
       - name: Spark Integration Test
         id: integrationTest
         run: |
-          ./gradlew --rerun-tasks -PskipTests -PtestMode=${{ matrix.test-mode }} -PjdkVersion=${{ matrix.java-version }} :spark-connector:spark-3.3:test --tests "com.datastrato.gravitino.spark.connector.integration.test.**"
-          ./gradlew --rerun-tasks -PskipTests -PtestMode=${{ matrix.test-mode }} -PjdkVersion=${{ matrix.java-version }} :spark-connector:spark-3.4:test --tests "com.datastrato.gravitino.spark.connector.integration.test.**"
-          ./gradlew --rerun-tasks -PskipTests -PtestMode=${{ matrix.test-mode }} -PjdkVersion=${{ matrix.java-version }} :spark-connector:spark-3.5:test --tests "com.datastrato.gravitino.spark.connector.integration.test.**"
+          if [ "${{ matrix.scala-version }}" == "2.12" ];then
```
Comment (caican00) [inline .github/workflows/spark-integration-test.yml:95]: if only support 2.12 in ci, seems unnecessary to add this check.

Comment (FANNG1) [reply .github/workflows/spark-integration-test.yml:95]: I prefer to keep this, maybe we need to support 2.13 latter

**Comment by FANNG1** [comment]
ci failed for `HDFSKerberosIT`, retrigger it

File: .github/workflows/spark-integration-test.yml (hunk 3 above)
Comment (caican00) [reply .github/workflows/spark-integration-test.yml:95]: ok.

File: settings.gradle.kts [hunk 4]
```diff
@@ -8,6 +8,10 @@ plugins {
 
 rootProject.name = "gravitino"
 
+val scalaVersion: String = gradle.startParameter.projectProperties["scalaVersion"]?.toString()
+  ?: settings.extra["defaultScalaVersion"]?.toString()
+  ?: "2.12"
```
Comment (caican00) [inline settings.gradle.kts:13]: Why not just define `defaultScalaVersion` directly in the `settings` file, it seems unnecessary to hardcode a `2.12` here.

Comment (FANNG1) [reply settings.gradle.kts]: `defaultScalaVersion` is defined in `gradle.properties`, removed `2.12`

**Comment by FANNG1** [comment]
@caican00 comments are addressed, please help to review again

**Comment by caican00** [comment]
> @caican00 comments are addressed, please help to review again

LGTM from my side.

**Comment by FANNG1** [comment]
@jerryshao do you have any comments?

**Comment by jerryshao** [comment]
> @jerryshao do you have any comments?

I don't have further comment.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**